### PR TITLE
Ported the turf-boolean-point-in-polygon function to Turf Swift.

### PR DESCRIPTION
**Pull request by SandyChapman**
Uses the identical algorithm as used in Turf.js.

https://github.com/Turfjs/turf/blob/e53677b0931da9e38bb947da448ee7404adc369d/packages/turf-boolean-point-in-polygon/index.ts

### CODE REVIEW

File: Sources/Turf/BoundingBox.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Foundation
+#if !os(Linux)
+import CoreLocation
+#endif
+
+public class BoundingBox {
+    
+    public init?(points: [CLLocationCoordinate2D]?) {
```
Comment (frederoni) [inline Sources/Turf/BoundingBox.swift]: nit: `public init?(from coordinates: [CLLocationCoordinate2D]?)`

perhaps an additional initializer:
`public init(_ northWest: CLLocationCoordinate2D, _ southEast: CLLocationCoordinate2D)`?

File: Sources/Turf/Polygon.swift [hunk 2]
```diff
@@ -59,3 +59,32 @@ extension Polygon {
...
+extension Polygon {
+    
+    /**
+     * Determines if the given point falls within the polygon and outside of its interior rings.
+     * The optional parameter `ignoreBoundary` will result in the method returning true if the given point
+     * lies on the boundary line of the polygon or its interior rings.
+     *
+     * Ported from: https://github.com/Turfjs/turf/blob/e53677b0931da9e38bb947da448ee7404adc369d/packages/turf-boolean-point-in-polygon/index.ts#L31-L75
+     */
+    public func contains(point: CLLocationCoordinate2D, ignoreBoundary: Bool = false) -> Bool {
```
Comment (frederoni) [inline Sources/Turf/Polygon.swift]: nit: `contains(_ coordinate: CLLocationCoordinate2D`

File: Sources/Turf/Polygon.swift [hunk 3]
```diff
@@ -59,3 +59,32 @@ extension Polygon {
...
+        let bbox = BoundingBox(points: self.coordinates.first)
```
Comment (frederoni) [inline Sources/Turf/Polygon.swift]: nit: self is inferred and can be dropped. (×2)

File: Sources/Turf/Ring.swift [hunk 4]
```diff
@@ -53,3 +53,44 @@ public struct Ring {
...
+
+extension Ring {
+    /**
+     * Determines if the given point falls within the ring.
+     * The optional parameter `ignoreBoundary` will result in the method returning true if the given point
+     * lies on the boundary line of the ring.
+     *
+     * Ported from: https://github.com/Turfjs/turf/blob/e53677b0931da9e38bb947da448ee7404adc369d/packages/turf-boolean-point-in-polygon/index.ts#L77-L108
+     */
+    public func contains(point: CLLocationCoordinate2D, ignoreBoundary: Bool = false) -> Bool {
```
Comment (frederoni) [inline Sources/Turf/Ring.swift]: nit: `contains(_ coordinate: CLLocationCoordinate2D`

File: Sources/Turf/BoundingBox.swift [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            .reduce((minLat: 0, maxLat: 0, minLon: 0, maxLon: 0)) { (result, coordinate) -> (minLat: Double, maxLat: Double, minLon: Double, maxLon: Double) in
+                let minLat = min(coordinate.latitude, result.0)
+                let maxLat = max(coordinate.latitude, result.1)
+                let minLon = min(coordinate.longitude, result.2)
+                let maxLon = max(coordinate.longitude, result.3)
+                return (minLat: minLat, maxLat: maxLat, minLon: minLon, maxLon: maxLon)
+        }
+    }
+    
+    public func contains(point: CLLocationCoordinate2D) -> Bool {
```
Comment (frederoni) [inline Sources/Turf/BoundingBox.swift]: nit: `contains(_ coordinate: `

**Comment by frederoni** [comment]
This PR partially fixes #50 and #33

**Comment by SandyChapman** [comment]
@frederoni : I believe all your feedback has been addressed. Can you please re-review and let me know if there are additional changes requested.

**Pull request merged**